src/wp-includes/media.php

```diff
@@ -5011,9 +5012,137 @@ function wp_getimagesize( $filename, array &$image_info = null ) {
 	 */
 	if ( 2 === func_num_args() ) {
 		// phpcs:ignore WordPress.PHP.NoSilencedErrors
-		return @getimagesize( $filename, $image_info );
+		return @_wp_get_image_size( $filename, $image_info );
 	} else {
 		// phpcs:ignore WordPress.PHP.NoSilencedErrors
-		return @getimagesize( $filename );
+		return @_wp_get_image_size( $filename );
 	}
 }
+
+/**
+ * Extract meta information about a webp file: width, height and type.
+ *
+ * @param [type] $filename Path to a WebP file.
+ * @return array $webp_info {
+ *     An array of WebP image information.
+ *
+ *     @type array $size {
+ *         @type int  $width  Image width.
+ *         @type int  $height Image height.
+ *         @type bool $type   The WebP type: one of 'lossy', 'lossless' or 'animated-alpha'.
+ *     }
+
+ */
+function wp_get_webp_info( $filename ) {
+	if ( ! 'image/webp' === wp_get_image_mime( $filename ) ) {
+		return array(
+			'width'  => false,
+			'height' => false,
+			'type'   => false,
+		);
+	}
```

Comment:
```suggestion
	$width  = false;
	$height = false;
	$type   = false;
	if ( ! 'image/webp' === wp_get_image_mime( $filename ) ) {
		return compact( 'width', 'height', 'type' );
	}
```
Is this a little cleaner?

Reply:
Nice!